app/javascript/leaflet/viewers/esri/dynamic_map_layer.js

```diff
@@ -0,0 +1,110 @@
+import { get, dynamicMapLayer, identifyFeatures } from "esri-leaflet";
+import linkifyHtml from "linkify-html";
+import LeafletViewerEsri from "../esri.js";
+
+export default class LeafletViewerEsriDynamicMapLayer extends LeafletViewerEsri {
+  // Override to parse between dynamic layer types
+  getEsriLayer() {
+    // Remove any trailing slash from endpoint url
+    this.data.url = this.data.url.replace(/\/$/, "");
+
+    // Get last segment from url
+    const pathArray = this.data.url.split("/");
+    const lastSegment = pathArray[pathArray.length - 1];
+
+    // If the last segment is an integer, slice and save as dynamicLayerId
+    if (Number(lastSegment) === parseInt(lastSegment, 10)) {
+      this.dynamicLayerId = lastSegment;
+      this.data.url = this.data.url.slice(0, -(lastSegment.length + 1));
+    }
+
+    get(this.data.url, {}, (error, response) => {
```

Comment:
probably be good to have some consistency in esri.js we have import { get as esriGet } from "esri-leaflet";

Reply:
I changed the other one; I think just `get` is fine